cmake/Modules/GeneratePkgConfig/generate-pkg-config.cmake.in

```diff
@@ -6,7 +9,7 @@ function (cmake_list_to_pkg_config _result _list _prefix)
 	# remove prefix from prefixed items
 	string(REGEX REPLACE "(^|;)(${_prefix})" "\\1" _tmp_list "${_tmp_list}")
 	# append 'prefix' to each element
-	string(REGEX REPLACE "([^;]+)" "${_prefix}\\1" _tmp_list "${_tmp_list}")
+	string(REGEX REPLACE "(^|;)([^-;]+)" "\\1${_prefix}\\2" _tmp_list "${_tmp_list}")
```

Comment:
Is this meant to prevent adding the prefix to any string starting with `-`?
It doesn't quite look like it it seems to just match two strings, and add a prefix to the second.

Reply:
I see, the first expression just matches the start of line *or* the `;`, not the full string

Reply:
While this is technically correct, the function you are changing is not aware of the "-" thing, but operates with the _prefix parameter value. Thus I would suggest to implement this in the following way:

1. Move all compiler parameters (i.e. list elements that begin with "-") from the list to another one.
2. Process the remaining list of files with the original `cmake_list_to_pkg_config`.
3. Append result to the first list with parameters.

Reply:
that would have the same effect as this though, right?

Reply:
at the very least, the comment should be updated to describe what's happening

Reply:
Yes, but it would be cleaner. Packing application logic in regular expressions smells bad, doesn't it?

Reply:
>     1. Move all compiler parameters (i.e. list elements that begin with "-") from the list to another one.
>     2. Process the remaining list of files with the original `cmake_list_to_pkg_config`.
>     3. Append result to the first list with parameters.

doing so may change link order, and this may be vital. consider:
- source command line is `-l<some_library_requires_pthread> -pthread`
- modified line: `-pthread -l<some_library_requires_pthread>`
[as result we will get "undefined reference" to pthread functions from ](https://stackoverflow.com/questions/45135/why-does-the-order-in-which-libraries-are-linked-sometimes-cause-errors-in-gcc)
so, original order should be preserved

Reply:
But -phread is a preprocessor and linker option, not a library.

Reply:
you are right, it doesn't matter in this case